cmd/create/machinepool/nodepool.go

```diff
@@ -540,7 +540,11 @@ func getSubnetFromAvailabilityZone(cmd *cobra.Command, r *rosa.Runtime, isAvaila
 		}
 		r.Reporter.Infof("There are several subnets for availability zone '%s'", availabilityZone)
 		interactive.Enable()
-		subnet := getSubnetFromUser(cmd, r, false, cluster)
+		subnet, err := getSubnetFromAvailabilityZone(cmd, r, isAvailabilityZoneSet, cluster)
```

Comment:
Hi @chenz4027,
is this recursion expected here? please kindly advise

Reply:
@davidleerh It is expected. The original error was that even if the user enters the availability zone, the subnets would include all subnets from all zones. Looking through the code, we were getting getSubnetFromUser instead of getSubnetFromAvailabilityZone. However, you are right it's introducing a recursion, let me see if I can revise it to avoid this behaviour

Reply:
@davidleerh Would you mind taking another look? Refactored